src/main/java/log_in/drivers/LogInDatabaseAccess.java

```diff
@@ -0,0 +1,126 @@
+package log_in.drivers;
+
+import entities.User;
+import com.opencsv.CSVReader;
+import log_in.use_case.LogInDsGateway;
+
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+
+public class LogInDatabaseAccess implements LogInDsGateway {
+    private final String userPath;
+
+    public LogInDatabaseAccess(String userPath){
+        this.userPath = userPath;
+    }
+
+
+    /**
+     * Method will call all emails in the database and check if the email given in the parameter
+     * is in the database.
+     * @param email email to be checked in the database.
+     * @return boolean of if the email exists in the database or not.
+     */
+    @Override
+    public boolean checkUserEmailExists(String email){
+        ArrayList<String> emails = this.getData(4);
+        for (int i = 0; i < emails.size(); i++){
+            if (emails.get(i).equals(email)){
+                return true;
+            } else {
+                i++;
```

Comment:
I don't think you need to increment i here, it will be done automatically by the for loop

Reply:
right, thanks